src/ClientHandle.cpp

```diff
@@ -1383,15 +1383,16 @@ void cClientHandle::HandleRightClick(int a_BlockX, int a_BlockY, int a_BlockZ, e
 
 		if (BlockHandler->IsUseable() && !m_Player->IsCrouched())
 		{
-			if (PlgMgr->CallHookPlayerUsingBlock(*m_Player, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace, a_CursorX, a_CursorY, a_CursorZ, BlockType, BlockMeta))
+			if (!PlgMgr->CallHookPlayerUsingBlock(*m_Player, a_BlockX, a_BlockY, a_BlockZ, a_BlockFace, a_CursorX, a_CursorY, a_CursorZ, BlockType, BlockMeta))
```

Comment:
That's exactly the crux of that change - if you return immediately, you're not leaving the "default" code below it a chance to run. I don't know if that's by design when it comes to plugins (i.e. should a plugin be able to block any interaction with a block, like placing a torch on it), but certainly when it comes to the non-plugin behaviour, my fix is necessary the way it is.